### PR TITLE
bug fix for disruption component

### PR DESCRIPTION
**Pull request by CoderOfTheLight**
Bug fixes for the disruption scorch component:
- physical.start_delay, dos.start_delay and dos.attack_duration are all separately defined within the cyberphysical functionality
- dos.attack_duration is checked to make sure it is not longer than run_duration
- the correct dos.attack_duration and dos.start_delay are added to _build_dos_cmd
- dos.attack_duration is given 1 minute instead of 1 second to finish in case it is longer than run_duration
- additional bugs mentioned in the component README

Testing:
 - 2 edge case tests - PASSED (dos.attack_duration too long)
 - 3 normal case tests - PASSED
 - 1 project related case test - PASSED

### CODE REVIEW

File: src/python/phenix_apps/apps/scorch/disruption/README.md [hunk 1]
```diff
@@ -72,3 +72,11 @@ components:
         log_path: /harmonie_scada.log
         python_path: /users/wwuser/appdata/local/programs/python/python38/python.exe
 ```
+
+## Notes and Considerations
+ * for a cyber_physical experiment, if overall run_time is shorter than combined dos.start_delay plus dos.attack_duration or physical.start_delay plus 5 minutes (whichever is longer), then phenix will fail because dos did not have enough time to export results or physical is not finished.
+ * for a cyber_physical experiment, if dos.start_delay is longer than the overall run_time, there is no no check shorten dos.start_delay in relation to dos.attack_duration. There is only a check to shorten dos.attack_duration.
```
Comment (GhostofGoes) [inline src/python/phenix_apps/apps/scorch/disruption/README.md:78]: grammar, `there is no no check shorten`

File: src/python/phenix_apps/apps/scorch/disruption/disruption.py [hunk 2]
```diff
@@ -174,45 +179,70 @@ def start(self):
...
-
-            remaining = run_duration - elapsed - start_delay
+            # run physical attack with start delay
+            physical_start_delay = float(self.metadata.physical.start_delay)
+            
+            # start delay
+            self.print(f"sleeping for {physical_start_delay} seconds")
+            sleep(physical_start_delay)
+
+            # attack duration
```
Comment (GhostofGoes) [inline src/python/phenix_apps/apps/scorch/disruption/disruption.py:189]: comment nitpick, it's not an attack

File: src/python/phenix_apps/apps/scorch/disruption/disruption.py [hunk 3]
```diff
@@ -174,45 +179,70 @@ def start(self):
...
+            if dos_run_duration > tmp_run_duration:
+                dos_run_duration = tmp_run_duration - dos_start_delay - 1.0
+                dos_run_duration = float(abs(dos_run_duration))
+            else:
+                dos_run_duration = dos_attack_duration
 
             # Kick off DoS process, have it sleep for number of seconds that puts it between line outage and load shedding
             # delay: 6.0 = 2.0 + 4.0 + 60.0 = 66.0
-            dos_start_delay = 2.0 + 4.0 + start_delay  # 66.0
+            #dos_start_delay = 2.0 + 4.0 + start_delay  # 66.0
```
Comment (GhostofGoes) [inline src/python/phenix_apps/apps/scorch/disruption/disruption.py:215]: delete old code, if we need it then it's in the git history on github

**Review by GhostofGoes — CHANGES_REQUESTED** [review]
Some minor nitpicks, otherwise LGTM!

**Review by GhostofGoes — APPROVED** [review]
